climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,849 @@
+            The default is {}.
+        """
+
+        self.unc_vars = unc_vars if unc_vars else {}
+        self.samples_df = samples if samples is not None else pd.DataFrame(
+            columns = self.param_labels)
+        self.sampling_method = None
+        self.metrics = metrics if metrics is not None else {}
+        self.sensitivity = sensitivity if sensitivity is not None else {}
+        self.check()
+
+
+    def check(self):
+        """
+        Check if the data variables are consistent
+
+        Returns
+        -------
+        check: boolean
+            True if data is consistent.
+
+        """
+        check = True
+        check &= (self.param_labels == self.samples_df.columns.to_list())
+        if not check:
+            raise ValueError("Parameter names from unc_vars do not "
+                             "correspond to parameters names of sample")
+        for metric, df_distr in self.metrics.items():
+            if not df_distr.empty:
+                check &= (len(df_distr) == self.n_samples)
+                if not check:
+                    raise ValueError(f"Metric f{metric} has less values than"
+                             " the number of samples {self.n_samples}")
+
+                if df_distr.isnull().values.any():
+                    LOGGER.warning("At least one metric evaluated to Nan for "
+                        "one cominbation of uncertainty parameters containend "
+                        "in sample. Note that the sensitivity analysis will "
+                        "then return Nan. "
+                        "See https://github.com/SALib/SALib/issues/237")
+        return check
+
+
+    @property
+    def n_samples(self):
+        """
+        The effective number of samples
+
+        Returns
+        -------
+        n_samples: int
+            effective number of samples
+
+        """
+
+        if isinstance(self.samples_df, pd.DataFrame):
+            return self.samples_df.shape[0]
+        return 0
+
+    @property
+    def param_labels(self):
+        """
+        Labels of all uncertainty parameters.
+
+        Returns
+        -------
+        param_labels: list of strings
+            Labels of all uncertainty parameters.
+
+        """
+        return list(self.distr_dict.keys())
+
+    @property
+    def distr_dict(self):
+        """
+        Dictionary of the distribution of all the parameters of all variables
+        listed in self.unc_vars
+
+        Returns
+        -------
+        distr_dict : dict( sp.stats objects )
+            Dictionary of all distributions.
+
+        """
+
+        distr_dict = dict()
+        for unc_var in self.unc_vars.values():
+            distr_dict.update(unc_var.distr_dict)
+        return distr_dict
+
+    @property
+    def problem(self):
+        """
+        The description of the uncertainty variables and their
+        distribution as used in SALib.
+        https://salib.readthedocs.io/en/latest/getting-started.html.
```

Comment:
```suggestion
        https://salib.readthedocs.io/en/latest/basics.html
```

Reply:
Already done in b2d7fce20822460ef5e8edaf317e8f0170f456db :)